Samples/NetworkMediaStreamer/Readme.md

```diff
@@ -0,0 +1,298 @@
+| | | |
+| ----------- | ----------- | -------- |
+| pPackethandler | Input pointer to ABI interface of EventHandler delegate that takes IBuffer pointer as an argument.| e.g. `auto handler = winrt::PacketHandler([](IInspectable sender, IBuffer args){ /*handle the rtp packet- send it over tcp etc.*/});` `pPacketHandler = handler.as<ABI::PacketHandler>().get()`|
+| pProtocol | Input pointer to string specifying the packetization format/protocol prefix | at present the default and only supported format is `L"rtp"`|
+| pParams | Input pointer to string containing extra parameters required to configure the client specific parameters in the format -  *param_name1=param_value1&param_name2=param_value2* | At present the only supported parameters are `ssrc` and `localrtpport`. e.g.-`L"ssrc=323454&localrtpport=5445"`. The default value for pParams is empty- an empty string  sets ssrc=0 and localrtpport is auto selected to an unused port.|
+
+
+`INetworkMediaStreamSink::RemoveTransportHandler(
+        ABI::PacketHandler* pPacketHandler)`
+Removed the specified custom transport handler delegate and stops calling the specified delegate for future packets.
+| | | |
+| ----------- | ----------- | -------- |
+| pPackethandler | Input pointer to ABI interface of EventHandler delegate to be removed.| see: `INetworkMediaStreamSink::AddTransportHandler`|
+
+
+
+`INetworkMediaStreamSink::GenerateSDP(
+        uint8_t* pBuf,
+        size_t maxSize,
+        LPCWSTR pDestination)`  
+Generated a Session Description Protocol buffer as per [RFC4566](https://tools.ietf.org/html/rfc4566)
+| | | |
+| ----------- | ----------- | -------- |
+| pBuf | Input pointer to the start of an allocated buffer to receive the SDP payload ||
+| maxSize | Input allocated size of the buffer pointed by pBuf| |
+| pDestination | Input pointer to a string containing destination ip address and port with a ':' separator. | e.g. `L"192.168.10.22:6554"` |
+
+
+`INetworkMediaStreamSink::Start(
+        MFTIME hnsSystemTime,
+        LONGLONG llClockStartOffset)`    
+This is used by the Sink to manage streaming clock states. Refer to [IMFClockStateSink::OnClockStart](https://docs.microsoft.com/en-us/windows/win32/api/mfidl/nf-mfidl-imfclockstatesink-onclockstart)
+
+`INetworkMediaStreamSink::Stop(MFTIME hnsSystemTime)`  
+This is used by the Sink to manage streaming clock states. Refer to [IMFClockStateSink::OnClockStop](https://docs.microsoft.com/en-us/windows/win32/api/mfidl/nf-mfidl-imfclockstatesink-onclockstop)
+
+`INetworkMediaStreamSink::Pause(MFTIME hnsSystemTime)`  
+This is used by the Sink to manage streaming clock states. Refer to [IMFClockStateSink::OnClockPause](https://docs.microsoft.com/en-us/windows/win32/api/mfidl/nf-mfidl-imfclockstatesink-onclockpause)
+
+`INetworkMediaStreamSink::Shutdown()`  
+This is used by the Sink to manage streaming clock states. Refer to [IMFMediaSink::Shutdown](https://docs.microsoft.com/en-us/windows/win32/api/mfidl/nf-mfidl-imfmediasink-shutdown)

```

Comment:
>This is used by the Sink to manage streaming clock states [](start = 0, length = 57)

Is the shutdown specific to clock stuff or is this comment inaccurate?

Reply:
good catch... its a Sink state

---
In reply to: [491136894](https://github.com/microsoft/Windows-Camera/pull/13#discussion_r491136894) [](ancestors = 491136894)